### ISSUE TITLE
appsync get query returns item, but list query doesn't

### ISSUE THREAD

**Issue by jesuscovam** [aws-amplify/amplify-js#9969]
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-js/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-js/discussions).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### JavaScript Framework

React Native

### Amplify APIs

GraphQL API

### Amplify Categories

api

### Environment information

<details>

```
# Put output below this line
System:
    OS: macOS 11.4
    CPU: (8) arm64 Apple M1
    Memory: 112.78 MB / 8.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 14.17.1 - ~/.nvm/versions/node/v14.17.1/bin/node
    Yarn: 1.19.0 - ~/.nvm/versions/node/v14.17.1/bin/yarn
    npm: 7.22.0 - ~/.nvm/versions/node/v14.17.1/bin/npm
  Browsers:
    Brave Browser: 102.1.39.111
    Chrome: 102.0.5005.61
    Safari: 14.1.1
  npmPackages:
    @gorhom/bottom-sheet: ^4.3.1 => 4.3.1
    @types/react: ~17.0.21 => 17.0.40
    autoprefixer: ^10.4.2 => 10.4.4
    axios: ^0.24.0 => 0.24.0
    graphql: ^14.7.0 => 14.7.0
    postcss: ^8.4.5 => 8.4.12
    react-native-calendars: ^1.1284.0 => 1.1284.0
  npmGlobalPackages:
    @aws-amplify/cli: 8.1.0
    @napi-rs/cli: 2.2.0
    aws-cdk: 2.23.0
    eas-cli: 0.52.0
    expo-cli: 5.4.8
    ganache-cli: 6.12.2
    ganache: 7.0.3
    npm: 7.22.0
    parcel: 2.0.0-rc.0
    react-devtools: 4.22.1
    truffle: 5.4.5
    typescript: 4.5.3
    yarn: 1.22.17

```

</details>

### Describe the bug

After a successfull mutation that creates an Item, I could do a get query with the ID and get 200, but If I do a list query the item is not returned in the data

### Expected behavior

I would expect that If I create an Item it will come on it's type list query.

### Reproduction steps

1. create the Item on the React Native App 
2. check the appsync console for the get Query with ID and it returns the Item
3. in the same appsync console do the list query and see that the item is not returning

### Code Snippet

```javascript
// Put your code below this line.

```

### Log output

<details>

```
// Put your logs below this line


```

</details>

### Additional information and screenshots

![Screen Shot 2022-06-06 at 10 48 35](https://user-images.githubusercontent.com/46403187/172428541-ede390f4-91da-42cf-8f3e-5e3959fb6e25.png)

**Comment by chrisbonifacio** [comment]
Hi @jesuscovam 👋 thanks for raising this issue! Unfortunately, I was not able to reproduce this behavior with my schema. I tried filtering items by id and did have the one record with a matching id returned.

<img width="979" alt="Screen Shot 2022-06-07 at 4 04 52 PM" src="https://user-images.githubusercontent.com/16952986/172472361-d95af497-dff3-4b9e-96e4-03312c82b175.png">


Could you share your schema so I can try reproducing more accurately?

**Comment by jesuscovam** [comment]
yes, this is the type
```
type CalendarDate
	@model
	@key(name: "byProfile", fields: ["profileID"])
{
	id: ID!
	date: AWSDate!
	status: DateStatus
	profileID: ID!
	blockedHours: [HourBlock]
	appointmentsSchedules: [AppointmentHour]
		@connection(keyName: "byCalendarDate", fields: ["id"])
}

enum DateStatus {
	BLOCKED
	OPEN
	SOME_HOURS_BLOCKED
	DAY_WITH_SERVICE
}

type HourBlock {
	id: ID!
	value: String!
	label: String!
	appointmentID: ID
	meetingID: ID
	service: String
	disabled: Boolean
}
```


This is the mutation in component
```
 const input: CreateCalendarDateInput = {
  date: newDate.date,
  profileID,
  status: DateStatus.BLOCKED,
  blockedHours: [],
 }

 const { data } = (await API.graphql(
  graphqlOperation(createCalendarDate, { input })
 )) as { data: CreateCalendarDateMutation }
```
I have more information, but is a private project and I have to ask before sharing it

**Comment by chrisbonifacio** [comment]
@jesuscovam Looks like you're using the V1 graphql-transformer in your schema since you're using `@key` and `@connection`. Is this intentional or do you want to migrate your schema over to V2?

I get this error when attempting to push your schema as it is to my API

<img width="1168" alt="Screen Shot 2022-06-09 at 10 42 52 AM" src="https://user-images.githubusercontent.com/16952986/172875009-40929fcf-30dc-4334-b368-36ded859a4b0.png">

If you don't mind, you could migrate your schema pretty easily by running the command `amplify migrate api`

For more info: https://docs.amplify.aws/cli/migration/transformer-migration/#changes-that-amplify-cli-will-auto-migrate-for-you

**Comment by jesuscovam** [comment]
Do we need to do this migration to get help on this issue?

Im really not looking forward to  deal with the possible ramifications of upgrading at the moment of the project, as we only need to fix a couple of issues including this for the launch of the apps.

**Comment by chrisbonifacio** [comment]
Could you share the VTL templates for your `listCalendarDates`'s resolver?

**Comment by jesuscovam** [comment]
Hi Sorry I haven't answered earlier.

I found my problem with Amplify + RN

**TLDR**
Don't collapse the RN bridge

I'm aware this shouldn't happen in any scenery, but the _corrupt_ items that were created in successful mutations and didn't 
return on the listQuery only happened when I did too much I/O on the app.

If I create the items without causing the APP to do too much I/O and flick, the items are created with the same function and return when the list query is called.

**Comment by chrisbonifacio** [comment]
> Hi Sorry I haven't answered earlier.
> 
> I found my problem with Amplify + RN
> 
> **TLDR** Don't collapse the RN bridge
> 
> I'm aware this shouldn't happen in any scenery, but the _corrupt_ items that were created in successful mutations and didn't return on the listQuery only happened when I did too much I/O on the app.
> 
> If I create the items without causing the APP to do too much I/O and flick, the items are created with the same function and return when the list query is called.

Interesting! Thank you for sharing this information with us. I'm going to put this out to the data team still in though for their awareness.

**Comment by jesuscovam** [comment]
> > Hi Sorry I haven't answered earlier.
> > I found my problem with Amplify + RN
> > **TLDR** Don't collapse the RN bridge
> > I'm aware this shouldn't happen in any scenery, but the _corrupt_ items that were created in successful mutations and didn't return on the listQuery only happened when I did too much I/O on the app.
> > If I create the items without causing the APP to do too much I/O and flick, the items are created with the same function and return when the list query is called.
> 
> Interesting! Thank you for sharing this information with us. I'm going to put this out to the data team for their awareness.
> 
> Could you provide some more info on how exactly you were creating items? How many at any given time, at what rate, etc?

Hi Sorry it took me so much time to reply

> how exactly you were creating items? 

https://user-images.githubusercontent.com/46403187/176754018-82bae0ea-9d61-4bc5-8b24-f287396f991c.mov

> How many at any given time
before this version of the app, I was allowing it to be more fast so maybe 1 Item x second, or maybe 2 item x second if possible.

and **this is what I changed to avoid having this issue**

I just made sure the function with the handler to create the item was an useCallback and made sure it was updating with the change of date selected.

**Comment by kevinxu3** [comment]
I also ran into this issue. @jesuscovam what was the solution you found?

**Comment by kevinxu3** [comment]
Can this ticket be reopened since I have this issue or should I open a new ticket?

**Comment by kevinxu3** [comment]
When calling:
```
query MyQuery {
  getUser(id: "cc7c803b-c6e6-4db0-9d22-9e7db3754eff") {
    id
    name
  }
  listUsers(filter: {or: {id: {eq: "cc7c803b-c6e6-4db0-9d22-9e7db3754eff"}}}) {
    items {
      id
      name
    }
  }
}
```
I get:
```
{
  "data": {
    "getUser": {
      "id": "cc7c803b-c6e6-4db0-9d22-9e7db3754eff",
      "name": "Demo User"
    },
    "listUsers": {
      "items": []
    }
  }
}
```
Instead of:
```
{
  "data": {
    "getUser": {
      "id": "cc7c803b-c6e6-4db0-9d22-9e7db3754eff",
      "name": "Demo User"
    },
    "listUsers": {
      "items": [
         {
            "id": "cc7c803b-c6e6-4db0-9d22-9e7db3754eff",
            "name": "Demo User"
         }
       ]
    }
  }
}
```
